P0009/P0009.bs

```diff
@@ -1805,28 +1796,16 @@ namespace std {
 namespace experimental {
 namespace fundamentals_v3 {
 
-template<class T>
 struct accessor_basic {
-  using value_type = T;
+  template<class T>
   using pointer = T*;
+  template<class T>
   using reference = T&;
-
-  // [mdspan.accessor.basic.ops], `accessor_basic` operations
-  constexpr reference operator()(pointer p, ptrdiff_t i) const noexcept;
 };
 
 }}}
 ```
 
-<br/>
-<b>26.7.�.2.1 `accessor_basic` operations [mdspan.accessor.basic.ops]</b>
-
-```c++
-constexpr reference operator()(pointer p, ptrdiff_t i) const noexcept;
```

Comment:
We think it is okish to delete that operator, but there are drawbacks and benefits:
- benefit: common case, pointer is just T* has now one less inlined function.
- drawback: requires pointer to be more often a wrapper object which contains all the stuff the accessor would contain, for example for the atomic accessor, pointer is not anymore T*. It needs to be some wrapper class which is convertible to T* and which has an [] operator.
- drawback: pointer must be templated on reference, since its operator [] must return that. 
- drawback: no clear separation of access and storage.
     - e.g. if I want to use some intrinsic to access the data in previous solution that intrinsic is in the operator of the accessor, now I write a little class containing a pointer and a [] operator, and the implementation of the [] operator contains the intrinsic

It seems this makes stuff more complex down the road. My main issue is that pointer[int] must return a reference, which means pointer and reference are not independent any more.

Reply:
Yep - not independent.  The *pointer* and *reference* should have a strong relationship.  The *pointer* is carried around as a member of mdspan and is handle to storage, the *reference* is an associated handle to accessing an element of that storage.  Should just be able to copy the storage handle and everything works - no extra wrappers.
See the prototype where `template<class T> using pointer = T *;` for the accessor_basic.